tf_agents/cogment_verse_tf_agents/reinforce/reinforce.py

```diff
@@ -93,31 +95,29 @@ def __loss(prob, actions, Q):
         loss = -log_prob * Q
         return tf.reduce_mean(loss)
 
-    def learn(self, batch, update_schedule=True):
+    def learn(self):
 
+        batch = self.sample_training_batch()
         Q = self.get_discounted_rewards(batch["rewards"])
         with tf.GradientTape() as tape:
             prob = self._model.model(batch["observations"], training=True)
-            prob = self.regularize_dist(prob)
             loss = self.__loss(prob, batch["actions"], Q)
         gradients = tape.gradient(loss, self._model.trainable_variables)
 
         self._optimizer.apply_gradients(zip(gradients, self._model.trainable_variables))
-
-        return {"loss": loss}
+        self.reset_replay_buffer()
```

Comment:
Should be "internalized" (at least mark as internal with the usual syntax)
```suggestion
        self._reset_replay_buffer()
```